### PR TITLE
Dshot API documentation

### PR DESCRIPTION
**Pull request by howels**
Adding content from https://brushlesswhoop.com/dshot-and-bidirectional-dshot/ with kind permission of the author https://github.com/stylesuxx

### CODE REVIEW

**Review by haslinghuis — DISMISSED** [review]

**Comment by howels** [comment]
> i don't feel this paragraph is necessary in the context of Betaflight documentation and can be replaced by something similar to "This documentation is reproduced with permission..." much like the statement currently at the end. (specifically usage of the "_**I**_" perspective off-putting for documentation, throughout.) image
> 
> the `dshot_bit-9ea340cf1820aa67d26415f231fce071.webp` image is rendered very large.

Of course, I will adapt this. For the image I think our page is scaling to fit the page width. Can we change that?

**Review by haslinghuis — APPROVED** [review]

File: docs/development/API/Dshot.md [hunk 1]
```diff
@@ -0,0 +1,448 @@
...
+
+From this example **we can conclude that with a PID loop frequency of 8kHz we can’t exhaust Dshot 300**, so there is no real reason to run Dshot600 - at least if your PID loop frequency is 8kHz or less.
+
+But this is actually not the whole truth, since the flight controller spaces out the frames and locks it to the PID loop frequency. Dshot frame generation thus always runs at PID loop rate - this on the other hand means, that if you are running really high PID loop frequencies, you also need to run a high Dshot version.
+
+> Should you for example run a 32kHz loop, the flight controller will send DHSOT frames every 31.25µs - meaning you have to run at least Dshot600 in order to keep up.
+
+## What is ESC Telemetry?
+
+In the section about Frames I mentioned a telemetry bit. The flight controller uses this bit to request telemetry information from the ESC.
```
Comment (nerdCopter) [inline docs/development/API/Dshot.md:196]:
```suggestion
In the section about Frames, it mentioned a telemetry bit. The flight controller uses this bit to request telemetry information from the ESC.
```

File: docs/development/API/Dshot.md [hunk 2]
```diff
@@ -0,0 +1,448 @@
...
+
+As mentioned in the [section above](#special-commands), bits 1-47 are reserved for special commands, some of which are used to request telemetry. Of those commands, 42-47 are related to telemetry - please reference the table to see which telemetry data you can query.
+
+### Transmission
+
+When the telemetry bit is set, the requested information is sent via a dedicated back-channel wire to the flight-controller. Multiple ESC’s can share the same wire. The protocol used is the KISS ESC telemetry protocol. And communicates the data via a single line back to the flight controller.
+
+**The frame size is a whopping 10 byte - 80 bit and is transmitted with a baudrate of 115200.**
+
+All telemetry data is transmitted in this frame. I do not want to go into further detail about ESC telemetry since it is not really part of Dshot. Detailed specifications can be found in an [rcGroups thread](http://www.rcgroups.com/forums/showatt.php?attachmentid=8524039&d=1450424877).
```
Comment (nerdCopter) [inline docs/development/API/Dshot.md:220]:
```suggestion
All telemetry data is transmitted in this frame. Detailed specifications can be found in an [rcGroups thread](http://www.rcgroups.com/forums/showatt.php?attachmentid=8524039&d=1450424877).
```

**Comment by nerdCopter** [comment]
> the `dshot_bit-9ea340cf1820aa67d26415f231fce071.webp` image is rendered very large.

@VitroidFPV , can you find a way to repair this?

File: docs/development/API/Dshot.md (hunk 1 above)
Comment (howels) [reply docs/development/API/Dshot.md:196]: Seems hard to say if this is better or worse.

File: docs/development/API/Dshot.md (hunk 2 above)
Comment (howels) [reply docs/development/API/Dshot.md]: Changed.

**Review by ctzsnooze — APPROVED** [review]

**Review by haslinghuis — APPROVED** [review]

**Pull request merged**